onetl/connection/file_connection/samba.py

```diff
@@ -0,0 +1,306 @@
+        try:
+            client.listShares()
+        except NotConnectedError:
+            return True
+        return False
+
+    def _close_client(self, client: SMBConnection) -> None:
+        self.client.close()
+
+    def _download_file(self, remote_file_path: RemotePath, local_file_path: LocalPath) -> None:
+        with open(local_file_path, "wb") as local_file:
+            self.client.retrieveFile(
+                self.share,
+                str(remote_file_path),
+                local_file,
+                show_progress=True,
+            )  # pysmb replaces '/', not works with <RemotePath> type
+
+    def _get_stat(self, path: RemotePath) -> RemotePathStat:
+        info = self.client.getAttributes(self.share, (os.fspath(path)))
+
+        if self.is_dir(os.fspath(path)):
+            return RemotePathStat(st_mode=stat.S_IFDIR)
+
+        return RemotePathStat(
+            st_size=info.file_size,
+            st_mtime=info.last_write_time,
+            st_uid=info.filename,
+        )
+
+    def _remove_file(self, remote_file_path: RemotePath) -> None:
+        self.client.deleteFiles(
+            self.share,
+            str(remote_file_path),
+        )  # pysmb replaces '/', not works with <RemotePath> type
+
+    def _create_dir(self, path: RemotePath) -> None:
+        path_parts = str(path).strip("/").split("/")
+        current_path_parts = []
+        for part in path_parts:  # create dirs sequentially as .createDirectory(...) cannot create nested dirs
+            current_path_parts.append(part)
+            current_path = "/".join(current_path_parts)
+            try:
+                self.client.createDirectory(self.share, current_path)
+            except OperationFailure:
+                pass
+
+    def _upload_file(self, local_file_path: LocalPath, remote_file_path: RemotePath) -> None:
+        with open(local_file_path, "rb") as file_obj:
+            self.client.storeFile(
+                self.share,
+                str(remote_file_path),
+                file_obj,
+            )  # pysmb replaces '/', not works with <RemotePath> type
+
+    def _rename_file(self, source: RemotePath, target: RemotePath) -> None:
+        self.client.rename(
+            self.share,
+            str(source),
+            str(target),
+        )  # pysmb replaces '/', not works with <RemotePath> type
+
+    def _remove_dir(self, path: RemotePath) -> None:
+        files = self.client.listPath(self.share, str(path))
+
+        for f in files:
+            if f.filename not in {".", ".."}:  # skip current and parent directory entries
+                full_path = f"{path}/{f.filename}"
+                if f.isDirectory:
+                    # recursively delete subdirectory
+                    self._remove_dir(full_path)
+                else:
+                    self.client.deleteFiles(self.share, full_path)
+
+        self.client.deleteDirectory(self.share, str(path))
```

Comment:
```suggestion
                full_path = path / f.filename
                if f.isDirectory:
                    # recursively delete subdirectory
                    self._remove_dir(full_path)
                else:
                    self.client.deleteFiles(self.share, os.fspath(full_path))

        self.client.deleteDirectory(self.share, os.fspath(path))
```
В текущем варианте `full_path` это строка, но мы передаем ее в `self._remove_dir`, который принимает на вход `RemotePath`. А `RemotePath(...) / str(...)` или `RemotePath(...).joinpath(...)` как раз возвращает объект типа `RemotePath`.

И нужно заменить однобуквенную переменную `f` на `file` или `item`